javascript/ql/src/semmle/javascript/dataflow/internal/FlowSteps.qll

```diff
@@ -14,7 +14,8 @@ import semmle.javascript.dataflow.Configuration
  */
 predicate shouldTrackProperties(AbstractValue obj) {
   obj instanceof AbstractExportsObject or
-  obj instanceof AbstractModuleObject
+  obj instanceof AbstractModuleObject or
+  obj.(DefinedCustomAbstractValue).shouldTrackProperties()
```

Comment:
I'd prefer for `DefinedCustomAbstractValue.shouldTrackProperties` to be strictly about type inference, not inter-procedural data flow, so I don't think I want this.